controllers/operator/certificate.go

```diff
@@ -251,12 +252,69 @@ func (r *AuthenticationReconciler) createV1CertificateIfNotPresent(authCR *opera
 	}
 }
 
+// addLabelIfMissing adds "manage-cert-rotation": "yes" label to the certificate if not exist
+func (r *AuthenticationReconciler) addLabelIfMissing(fieldsList []*reconcileCertificateFields) (fn subreconciler.Fn) {
+	return func(ctx context.Context) (result *ctrl.Result, err error) {
+		reqLogger := ctrl.LoggerFrom(ctx)
+
+		allV1CertReconcilers := make([]subreconciler.Fn, 0)
+		for _, fields := range fieldsList {
+			allV1CertReconcilers = append(allV1CertReconcilers, r.updateCertWithLabel(fields))
+		}
+		results := []*ctrl.Result{}
+		errs := []error{}
+		for _, reconcileV1Cert := range allV1CertReconcilers {
+			result, err = reconcileV1Cert(ctx)
+			results = append(results, result)
+			errs = append(errs, err)
+		}
+
+		result, err = ctrlcommon.ReduceSubreconcilerResultsAndErrors(results, errs)
+		if subreconciler.ShouldContinue(result, err) {
+			reqLogger.Info("No certificates to be labeled")
+		} else if subreconciler.ShouldRequeue(result, err) && err == nil {
+			reqLogger.Info("Certificates were labeled; requeueing")
+		} else if err != nil {
+			reqLogger.Info("Encountered an issue while trying to label certificates")
+		}
+
+		return
+	}
+}
+
+func (r *AuthenticationReconciler) updateCertWithLabel(fields *reconcileCertificateFields) subreconciler.Fn {
+	return func(ctx context.Context) (result *ctrl.Result, err error) {
+		reqLogger := ctrl.LoggerFrom(ctx, "Certificate.Name", fields.Name, "Certificate.Namespace", fields.Namespace)
```

Comment:
Why are you calling `ctrl.LoggerFrom` instead of `logf.FromContext`?

Reply:
I see - weird. I've been using logf instead for most of the logging, like [here](https://github.com/IBM/ibm-iam-operator/pull/996/files#diff-728e9d77a68bd494cb2f0cde922a584a4b4061bc9bd7462f2af749543d82d38dL43).